scaladoc/src/dotty/tools/scaladoc/snippets/FlexmarkSnippetProcessor.scala

```diff
@@ -17,40 +17,85 @@ object FlexmarkSnippetProcessor:
-              case Right(flags) => Some(flags)
-              case Left(error) =>
-                report.warning(
-                  s"""|Error occured during parsing flags in snippet:
-                      |$error""".stripMargin
-                )
-                None
-            })
-        val snippetCompilationResult = cf(snippet, lineOffset, argOverride) match {
-          case result@Some(SnippetCompilationResult(wrapped, _, _, _)) if debug =>
+        val argOverride = info
+          .find(_.startsWith("sc:"))
+          .map(_.stripPrefix("sc:"))
+          .map(SCFlagsParser.parse)
+          .flatMap(_ match {
+            case Right(flags) => Some(flags)
+            case Left(error) =>
+              report.warning(
+                s"""|Error occured during parsing flags in snippet:
+                    |$error""".stripMargin
+              )
+              None
+          })
+        val id = info
+          .find(_.startsWith("sc-id:"))
+          .map(_.stripPrefix("sc-id:"))
+
+        val snippetImports = info
+          .find(_.startsWith("sc-compile-with:"))
+          .toList
+          .map(_.stripPrefix("sc-compile-with:"))
+          .flatMap(_.split(","))
+          .flatMap { id =>
+            val snippet = snippetMap.get(id)
+            if snippet.isEmpty then
+              report.warning(
+                s"""|Error occured during parsing compile-with in snippet:
+                    |Snippet with id: $id not found.
+                    |Remember that you cannot use forward reference to snippets""".stripMargin
+              )
+            snippet
+          }.mkString("\n")
+
+        val snippet = node.getContentChars.toString
+
+        extension (n: mdu.Node)
+          def setContentString(str: String): Unit =
             val s = sequence.BasedSequence.EmptyBasedSequence()
-              .append(wrapped.snippet)
+              .append(str)
               .append(sequence.BasedSequence.EOL)
             val content = mdu.BlockContent()
             content.add(s, 0)
             node.setContent(content)
+
+        val importedSection = if snippetImports.nonEmpty then
+          s"""|//{i
+              |$snippetImports
+              |//i}
+              |""".stripMargin
+        else ""
+        val fullSnippet = importedSection + snippet
+        val snippetCompilationResult = cf(fullSnippet, lineOffset, argOverride) match {
+          case result@Some(SnippetCompilationResult(wrapped, _, _, _)) if debug =>
+            node.setContentString(wrapped.snippet)
+            result
+          case result =>
+            node.setContentString(fullSnippet)
             result
-          case result => result
         }
 
         node.insertBefore(new ExtendedFencedCodeBlock(node, snippetCompilationResult))
         node.unlink()
-      }
+        id.fold(snippetMap)(id =>
+          val entry = (id, Seq(snippetImports, snippet).mkString("\n"))
+          snippetMap + entry
+        )
+      } else snippetMap
```

Comment:
Is there a reference documentation of all the changes implemented here? That would make the review easier.

Reply:
As for now there's no documentation. I'm gonna add it in this PR